threebot.go

```diff
@@ -0,0 +1,167 @@
+package threebot
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/coredns/coredns/plugin"
+	"github.com/miekg/dns"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strings"
+)
+
+type Threebot struct {
+	Next           plugin.Handler
+	Ttl            uint32
+	Zones          []string
+	Explorers	   []string
+}
+
+type Zone struct {
+	Name      string
+	Locations map[string]Record
+}
+
+type Record struct {
+	A     []A_Record `json:"a,omitempty"`
+	AAAA  []AAAA_Record `json:"aaaa,omitempty"`
+	CNAME []CNAME_Record `json:"cname,omitempty"`
+
+}
+
+type A_Record struct {
+	Ttl uint32 `json:"ttl,omitempty"`
+	Ip  net.IP `json:"ip"`
+}
+
+type AAAA_Record struct {
+	Ttl uint32 `json:"ttl,omitempty"`
+	Ip  net.IP `json:"ip"`
+}
+
+type CNAME_Record struct {
+	Ttl  uint32 `json:"ttl,omitempty"`
+	Host string `json:"host"`
+}
+
+func (threebot *Threebot) A(name, z string,  record *Record) (answers, extras []dns.RR) {
+	for _, a := range record.A {
+		if a.Ip == nil {
+			continue
+		}
+		r := new(dns.A)
+		r.Hdr = dns.RR_Header{Name: name, Rrtype: dns.TypeA,
+			Class: dns.ClassINET, Ttl: threebot.minTtl(a.Ttl)}
+		r.A = a.Ip
+		answers = append(answers, r)
+	}
+	return
+}
+
+func (threebot Threebot) AAAA(name, z string,  record *Record) (answers, extras []dns.RR) {
+	for _, aaaa := range record.AAAA {
+		if aaaa.Ip == nil {
+			continue
+		}
+		r := new(dns.AAAA)
+		r.Hdr = dns.RR_Header{Name: name, Rrtype: dns.TypeAAAA,
+			Class: dns.ClassINET, Ttl: threebot.minTtl(aaaa.Ttl)}
+		r.AAAA = aaaa.Ip
+		answers = append(answers, r)
+	}
+	return
+}
+
+func (threebot *Threebot) CNAME(name, z string,  record *Record) (answers, extras []dns.RR) {
+	for _, cname := range record.CNAME {
```

Comment:
Do you populate the AAA field anywhere in the code ? cause it seems this will always be empty.
Is that ok ?